app-drawer/app-drawer.html

```diff
@@ -535,6 +538,39 @@
         this.fire('app-drawer-overlay-changed', { overlay: this.opened && !this.persistent });
       },
 
+      _openedChanged: function(opened) {
+        var evtName = this.opened ? 'opened' : 'closed';
+
+        this.fire('app-drawer-' + evtName);
+      },
+
+      _onTransitionEnd: function(evt) {
+        var opened = this.opened;
+        var evtName = this.opened ? 'opened' : 'closed';
+
+        this.fire('app-drawer-transitioned-' + evtName);
```

Comment:
I see - I think the best thing to do would be to update the code for the `app-drawer-overlay-changed` event such that it relies on contentContainer instead of the scrim. We need to be careful though since we only want to fire the event once - there is no transitionend if you drag the drawer all the way open then let go, but there is when you drag it all the way closed since `translateX(-100%)` != `translateX(-256px)`. That's why there's:

```javascript
        // Trigger drawer overlay changed now since there will be no transitionend event.
        if (isInEndState) {
          this._drawerOverlayChanged();
        }
```

which is fine right now since it's the scrim, but that will have to change if it's for contentContainer.

I'll take a look at implementing this myself, but I just want clarification that this is what you need. Thanks.